bra-ket and plus/minus for Latex parser

bhpayne opened the pull request:
#### References to other Issues or PRs
https://github.com/sympy/sympy/issues/19854


#### Brief description of what is fixed or changed
@iammosespaulr  added tokens  plus/minus and for  bra-ket Dirac notation

#### Other comments
In collaboration with @msgoff and @iammosespaulr


#### Release Notes


<!-- BEGIN RELEASE NOTES -->
* parsing
   * Improved Latex support for reading expressions into SymPy
<!-- END RELEASE NOTES -->

bhpayne committed:
added plus/minus and bra/ket tokens

bhpayne committed:
use the bra/ket tokens

bhpayne committed:
added import but no test yet

iammosespaulr commented on sympy/parsing/latex/_parse_latex_antlr.py:
```diff
@@ -319,7 +319,13 @@ def convert_atom(atom):
         text = rule2text(atom.mathit().mathit_text())
         return sympy.Symbol(text)
-
+    elif atom.bra():
```
you gotta add this import at the top 
```python3
from sympy.physics.quantum.state import Bra, Ket
```

bhpayne committed:
removed spaces 

Travis-CI indicated spaces are unacceptable

bhpayne committed:
added a test

Travis-CI was complaining that Bra and Ket were imported but not used

iammosespaulr commented on sympy/parsing/tests/test_latex.py:
```diff
@@ -143,6 +144,8 @@ def test_import():
      Integral(3*_Pow(theta, -1), theta)),
     ("\\int \\frac{1}{x} + 1 dx", Integral(_Add(_Pow(x, -1), 1), x)),
     ("x_0", Symbol('x_{0}')),
+    ("\langle x |", Bra('x')),
```
```suggestion
    ("\\langle x |", Bra('x')),
```

iammosespaulr commented on sympy/parsing/tests/test_latex.py:
```diff
@@ -143,6 +144,8 @@ def test_import():
     ("\\int \\frac{1}{x} + 1 dx", Integral(_Add(_Pow(x, -1), 1), x)),
     ("x_0", Symbol('x_{0}')),
+    ("\langle x |", Bra('x')),
+    ("| x \rangle", Ket('x')),
```
```suggestion
    ("| x \\rangle", Ket('x')),
```

bhpayne committed:
forgot the import statement

bhpayne committed:
Update sympy/parsing/tests/test_latex.py

Co-authored-by: Moses Paul R <iammosespaulr@gmail.com>

bhpayne committed:
Update sympy/parsing/tests/test_latex.py

Co-authored-by: Moses Paul R <iammosespaulr@gmail.com>

bhpayne commented:
fixed the issues that @iammosespaulr pointed out

bhpayne committed:
antlr build artifacts

bhpayne committed:
corrected version of Antlr from 4.8 to 4.7.2

bhpayne committed:
removed left[ and \right] since test was failing

these are not vital for Dirac notation

LEFT_BRACKET: '\\left[';
RIGHT_BRACKET: '\\right]';

bhpayne committed:
corrected version of Antlr from 4.8 to 4.7.2

bhpayne committed:
remove []

bhpayne committed:
Merge branch 'braket-patch' of https://github.com/bhpayne/sympy into braket-patch

czgdp1807 commented:
@iammosespaulr Is it good to go?

czgdp1807 commented:
@bhpayne Could you please resolve the conflicts if you are interested in continuing this? 
@iammosespaulr @asmeurer Please review.
